descwl_coadd/coadd.py

```diff
@@ -1,1236 +1,639 @@
-                # self.weight,
-            ],
-        )
+        var = np.median(corrected_var)
+    else:
+        var = np.median(variance[use])
 
-    def _make_coadds(self):
-        """
-        make warps and coadds for images and noise fields
-        """
-        image_data = self._make_warps(
-            exps=self.exps,
-            bbox=self.coadd_bbox,
-            wcs=self.coadd_wcs,
-        )
-        # note backwards from C/python order
-        psf_nx, psf_ny = self.psf_dims
-        psf_bbox = geom.Box2I(
-            geom.Point2I(0, 0),
-            geom.Point2I(psf_nx-1, psf_ny-1),
-        )
+    noise_image = rng.normal(scale=np.sqrt(var), size=signal.shape)
 
-        psf_data = self._make_warps(
-            exps=self.psf_exps,
-            bbox=psf_bbox,
-            wcs=self.coadd_psf_wcs,
-        )
+    ny, nx = signal.shape
+    nmimage = afw_image.MaskedImageF(width=nx, height=ny)
+    assert nmimage.image.array.shape == (ny, nx)
 
-        noise_data = self._make_warps(
-            exps=self.noise_exps,
-            bbox=self.coadd_bbox,
-            wcs=self.coadd_wcs,
-        )
+    nmimage.image.array[:, :] = noise_image
+    nmimage.variance.array[:, :] = var
+    nmimage.mask.array[:, :] = exp.mask.array[:, :]
 
-        # we need the weights in the coadds to be the same
-        # so we replace them here
-        psf_data["weights"] = image_data["weights"]
-        noise_data["weights"] = image_data["weights"]
+    noise_exp = afw_image.ExposureF(nmimage)
+    noise_exp.setPsf(exp.getPsf())
+    noise_exp.setWcs(exp.getWcs())
+    noise_exp.setFilterLabel(exp.getFilterLabel())
+    noise_exp.setDetector(exp.getDetector())
 
-        # now we coadd
-        self.coadd_psf_exp = self._make_coadd(**psf_data)
-        pimage = self.coadd_psf_exp.image.array
-        wbad = np.where(~np.isfinite(pimage))
-        if wbad[0].size == pimage.size:
-            raise ValueError('no good pixels in the psf')
-        if wbad[0].size > 0:
-            self.log.info('zeroing %d bad psf pixels' % wbad[0].size)
-            pimage[wbad] = 0.0
+    return noise_exp, var
 
-        self.coadd_exp = self._make_coadd(**image_data)
-        self.coadd_exp.setPsf(make_stack_psf(pimage))
 
-        self.coadd_noise_exp = self._make_coadd(**noise_data)
-        self.coadd_noise_exp.setPsf(make_stack_psf(pimage))
+def get_psf_exp(exp, coadd_wcs, var):
+    """
+    create a psf exposure to be coadded, rendered at the
+    position in the exposure corresponding to the center of the
+    coadd
 
-    def _make_warps(self, *, exps, bbox, wcs):
-        """
-        make the warp images
+    Parameters
+    ----------
+    exp: afw_image.ExposureF
+        The exposure
+    coadd_wcs: DM wcs
+        The wcs for the coadd
+    var: float
+        The variance to set in the psf variance map
+
+    Returns
+    -------
+    psf ExposureF
+    """
+    coadd_sky_orig = coadd_wcs.getSkyOrigin()
```

Comment:
I guess this won't handle patches yet? I think each patch shares the overall WCS of the tract and is simple a subset of it?

Reply:
could be a bug

Reply:
I think this was a bug, pushed what I think is a fix